rust/serving/src/config.rs

```diff
@@ -1,40 +1,24 @@
+use std::env;
 use std::fmt::Debug;
-use std::path::Path;
-use std::{env, sync::OnceLock};
 
 use async_nats::rustls;
 use base64::prelude::BASE64_STANDARD;
 use base64::Engine;
-use config::Config;
 use rcgen::{generate_simple_self_signed, Certificate, CertifiedKey, KeyPair};
 use serde::{Deserialize, Serialize};
-use tracing::info;
 
 use crate::Error::ParseConfig;
-use crate::{Error, Result};
+use crate::Result;
 
-const ENV_PREFIX: &str = "NUMAFLOW_SERVING";
 const ENV_NUMAFLOW_SERVING_SOURCE_OBJECT: &str = "NUMAFLOW_SERVING_SOURCE_OBJECT";
 const ENV_NUMAFLOW_SERVING_JETSTREAM_URL: &str = "NUMAFLOW_ISBSVC_JETSTREAM_URL";
 const ENV_NUMAFLOW_SERVING_JETSTREAM_STREAM: &str = "NUMAFLOW_SERVING_JETSTREAM_STREAM";
 const ENV_NUMAFLOW_SERVING_STORE_TTL: &str = "NUMAFLOW_SERVING_STORE_TTL";
-
-pub fn config() -> &'static Settings {
-    static CONF: OnceLock<Settings> = OnceLock::new();
-    CONF.get_or_init(|| {
-        let config_dir = env::var("CONFIG_PATH").unwrap_or_else(|_| {
-            info!("Config directory is not specified, using default config directory: './config'");
-            String::from("config")
-        });
-
-        match Settings::load(config_dir) {
-            Ok(v) => v,
-            Err(e) => {
-                panic!("Failed to load configuration: {:?}", e);
-            }
-        }
-    })
-}
+const ENV_NUMAFLOW_SERVING_HOST_IP: &str = "NUMAFLOW_SERVING_HOST_IP";
+const ENV_NUMAFLOW_SERVING_APP_PORT: &str = "NUMAFLOW_SERVING_APP_LISTEN_PORT";
+const ENV_NUMAFLOW_SERVING_JETSTREAM_USER: &str = "NUMAFLOW_ISBSVC_JETSTREAM_USER";
+const ENV_NUMAFLOW_SERVING_JETSTREAM_PASSWORD: &str = "NUMAFLOW_ISBSVC_JETSTREAM_PASSWORD";
+const ENV_NUMAFLOW_SERVING_AUTH_TOKEN: &str = "NUMAFLOW_SERVING_AUTH_TOKEN";
```

Comment:
create a mod called serving and move serving items there? config is getting bloated

Reply:
This is the `serving::config` module. As of now, it not that big. I think single module is fine for now.